.helm/ecamp3/templates/print_deployment.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "print.name" . }}
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.print.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "app.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "app.selectorLabels" . | nindent 8 }}
+      annotations:
+        # This deployment should be restarted whenever either the configmap or the secrets change
+        # because the container depends on environment variables from these places during startup
+        checksum/config: {{ include (print $.Template.BasePath "/print_configmap.yaml") . | sha256sum }}
+        checksum/secrets: {{ include (print $.Template.BasePath "/print_secrets.yaml") . | sha256sum }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "app.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}-print
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.print.image.repository }}:{{ .Values.print.image.tag | default .Values.imageTag }}"
+          imagePullPolicy: {{ .Values.print.image.pullPolicy }}
+          ports:
+            - name: print-http
+              containerPort: 3003
+              protocol: TCP
+          readinessProbe:
+            httpGet:
+              scheme: HTTP
+              path: /
+              port: 3003
+            initialDelaySeconds: 10
+            periodSeconds: 5
+          livenessProbe:
```

Comment:
Something seems to be off here: Kubernetes is currently restarting the print pods very often due to failed readiness probe and failed liveness probe.

Reply:
Aha, I see. The probes simply send an http request to `http://<print-pod-ip>:3003`, which often doesn't get any response (but sometimes it does? Because for short periods of time, the pod is recognized as alive and ready).

Do you have any idea what the problem could be? Do we have to tell the nuxt server to listen on any ip address instead of specific hostnames? Do you know of any better way to check whether a nuxt server is running and ready for network traffic? There are also other options than a http request: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

Reply:
Not sure what the problem is here. From the outside, the print server seems to be accessible and delivers a proper 200 response.

In .docker-hub/print/Dockerfile nuxt seems to be configured to listen on any hostname (ENV NUXT_HOST 0.0.0.0). Maybe we could try below to set the environment variable again:
```
- name: NUXT_PORT
  value: "3003"
- name: NUXT_HOST
  value: "0.0.0.0"
```

Depending on what the root URL / will respond in the future, we might want to configure the healthcheck on a different URL (e.g. avoiding printing a PDF everytime the healtch check runs).
Something like https://github.com/fukuiretu/nuxt-healthcheck. But we can implement this, once we decided/discussed on how to move forward with printing in general.

Reply:
If I understand the [official documentation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes) and this [blog post](https://blog.colinbreck.com/kubernetes-liveness-and-readiness-probes-how-to-avoid-shooting-yourself-in-the-foot/) correctly, the default timeout of the probe is 1s only. Maybe nuxt takes more than 1s from time to time (not sure what is currently processed under root URL anyway).

Should we try to increase the timeout until we have a dedicated & fast healthcheck endpoint?
```
timeoutSeconds: 10
periodSeconds: 30
```

Reply:
It's worth a try. I set the values like you suggested, and so far there have been no restarts of the print service. Let's keep monitoring this a little, and then introduce the faster healthcheck some time in the future.